packages/sanity/src/structure/panes/document/DocumentPaneProvider.tsx

```diff
@@ -567,6 +529,47 @@ export const DocumentPaneProvider = memo((props: DocumentPaneProviderProps) => {
     [formStateRef],
   )
 
+  const handleFocus = useCallback(
+    (nextFocusPath: Path) => {
+      if (!deepEquals(focusPathRef.current, nextFocusPath)) {
+        setFocusPath(nextFocusPath)
+        focusPathRef.current = nextFocusPath
+        openPathRef.current = nextFocusPath
+        onFocusPath?.(nextFocusPath)
+      }
+
+      presenceStore.setLocation([
+        {
+          type: 'document',
+          documentId,
+          path: nextFocusPath,
+          lastActiveAt: new Date().toISOString(),
+        },
+      ])
+    },
+    [documentId, onFocusPath, presenceStore, setFocusPath],
+  )
+
+  const handleBlur = useCallback(
+    (blurredPath: Path) => {
+      if (disableBlurRef.current) {
+        return
+      }
+
+      setFocusPath(EMPTY_ARRAY)
+      setOpenPath(blurredPath)
```

Comment:
Why do we set the blurred path to be open here? This doesn't seem right, and is causing issues with opening (existing) text annotations in the PTE.

Reply:
Sometimes some initial `openPath` state is set: via a URL when using Structure tool, or anytime you click on an element in Presentation.

This state is never changed, which means that if the initial `openPath` references some block in the the PTE, it will force the scroll position back to that initial block as you try to move the cursor to a different one.

So as I understood it, the `openPath` state needs to be unset/reset somehow when e.g. clicking outside of the PTE to fix this scroll issue, hence the blur event.

I see now that Annotation calling `onPathFocus` with an empty path will trigger a blur event, which with this change causes the annotation to immediately close. Have I understood correctly?